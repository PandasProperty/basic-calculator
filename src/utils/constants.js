export const ROLES = {
    OPERAND: 'operand',
    OPERATOR: 'operator',
    RESET: 'reset',
    INPUT: 'input'
};

export const EQUALS = '='; 