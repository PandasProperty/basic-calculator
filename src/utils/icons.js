import {
    faPlus, faMinus, faPercentage,faTimes, faEquals
} from '@fortawesome/free-solid-svg-icons';

export default {
    plus: faPlus,
    minus: faMinus,
    percentage: faPercentage,
    times: faTimes,
    equals: faEquals
}