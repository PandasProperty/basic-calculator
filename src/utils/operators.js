export default [{
    value: '/',
    icon: 'percentage'
}, {
    value: '*',
    icon: 'times'
}, {
    value: '+',
    icon: 'plus'
}, {
    value: '-',
    icon: 'minus'
}, {
    value: '=',
    icon: 'equals'
}];