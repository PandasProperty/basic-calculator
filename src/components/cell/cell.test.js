import React from 'react';
import Cell from '.';
import { noop } from 'lodash';
import renderer from 'react-test-renderer';
import { ROLES } from '../../utils/constants';
import operators from '../../utils/operators';

describe('Cell', () => {
    
    it('Cell should render correctly for an operand', () => {
        const component = renderer.create(
            <Cell size={2} onClick={noop} content={0} role={ROLES.OPERAND} />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Cell should render correctly for an operator', () => {
        const component = renderer.create(
            <Cell onClick={noop} content={operators[1]} role={ROLES.OPERATOR} />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Cell should render correctly for a reset', () => {
        const component = renderer.create(
            <Cell size={3} onClick={noop} content='C' role={ROLES.RESET} />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Cell should render correctly for an input', () => {
        const component = renderer.create(
            <Cell size={4} content='2 * 4 + 1' role={ROLES.INPUT} />
        );

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});