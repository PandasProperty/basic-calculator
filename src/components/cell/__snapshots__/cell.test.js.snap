// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell Cell should render correctly for a reset 1`] = `
<div
  className="cell__container cell__container--3"
>
  <div
    className="cell__content cell__content--reset"
    onClick={[Function]}
  >
    C
  </div>
</div>
`;

exports[`Cell Cell should render correctly for an input 1`] = `
<div
  className="cell__container cell__container--4"
>
  <div
    className="cell__content cell__content--input"
    onClick={[Function]}
  >
    2 * 4 + 1
  </div>
</div>
`;

exports[`Cell Cell should render correctly for an operand 1`] = `
<div
  className="cell__container cell__container--2"
>
  <div
    className="cell__content cell__content--operand"
    onClick={[Function]}
  >
    0
  </div>
</div>
`;

exports[`Cell Cell should render correctly for an operator 1`] = `
<div
  className="cell__container cell__container--1"
>
  <div
    className="cell__content cell__content--operator"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-times fa-w-11 "
      data-icon="times"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 352 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
